batch/batch/cloud/azure/instance_config.py

```diff
@@ -75,13 +75,16 @@ def __init__(
 
         machine_type_parts = azure_machine_type_to_parts(self._machine_type)
         assert machine_type_parts
+        self.machine_type_parts = machine_type_parts
         self._worker_type = machine_type_parts.family
         self.cores = machine_type_parts.cores
 
     def worker_type(self) -> str:
         return self._worker_type
 
     def cores_mcpu_to_memory_bytes(self, mcpu: int) -> int:
+        if self.job_private:
```

Comment:
Injecting this code in here feels fishy: it checks a field of the object and conditionally ignores the function input, which violates my assumptions on what the return value means given the function name. That tells me that maybe in that condition (job_private) we should never have called this function in the first place. Is it possible to lift this check out of this method further up the call stack?